src/fava/templates/macros/_commodity_macros.html

```diff
@@ -0,0 +1,20 @@
+{% macro render_currency(currency) %}
+{% set alias = ledger.commodities.get(currency).meta.get('name') if ledger is defined else none %}
+<span title="{{ alias | default(currency) }}">{{ currency }}</span>
+{% endmacro %}
+
+{% macro render_amount(amount, class="num") %}
+{% if amount is none or amount.number is none %}
+  <span class="{{ class }}"><span>
+{% else %}
+  {% set commodity = ledger.commodities.get(amount.currency) if ledger is defined else none %}
```

Comment:
It might make sense to extract this and and the following lines ("does this currency have a name") into a method on the FavaLedger class, say, `currency_name(cur: str) -> Optional[str]`. That would also make it more testable to ensure it works correctly in all different cases (no currency directive/currency directive without `name` metadata/currency directive with `name` metadata)